molecule/default/tests/test_reboot_devices.py

```diff
@@ -0,0 +1,58 @@
+import os
+import re
+import pytest
+import testinfra.utils.ansible_runner
+
+
+testinfra_hosts = testinfra.utils.ansible_runner.AnsibleRunner(
+    os.environ['MOLECULE_INVENTORY_FILE']).get_hosts('dashboard_hosts')
+
+pre_cmd = "sudo bash -c \"source /root/openrc; "
+
+
+@pytest.mark.jira('asc-228')
+@pytest.mark.skip(reason='add skipped tests for asc-301')
+def test_to_reboot_nova_compute(host):
+    cmd = pre_cmd + "ansible -m shell nova_compute -a \'reboot\' \""
```

Comment:
I am not a fan of the "ansible within ansible" pattern. This seems like asking for trouble. I will not block merging in the short term because this is a verbatim implementation of the deploy validation. However, I would prefer a __TODO__ or __FIXME__ to note that this pattern should be replaced by using the molecule test files to dictate the scope of the host groups and use molecule to leverage ansible behind the scenes to run basic shell commands.

Reply:
I agree, will add a ticket to find out how to filter hosts from the list of hosts from testinfra get_hosts()